src/processor/map/map_gds.cpp

```diff
@@ -14,6 +15,19 @@ using namespace kuzu::storage;
 namespace kuzu {
 namespace processor {
 
+LogicalSemiMasker* PlanMapper::findSemiMaskerInPlan(LogicalOperator* logicalOperator) {
```

Comment:
I think we can check the child of LogicalDummySink must be LogicalSemiMasker instead of doing a full top-down traversal?